accounts/abi/bind/bind.go

```diff
@@ -594,6 +596,20 @@ func decapitalise(input string) string {
 	return strings.ToLower(goForm[:1]) + goForm[1:]
 }
 
+// convertToNil converts any type to its proper nil form.
+func convertToNil(input abi.Type) string {
+	switch input.T {
+	case abi.IntTy, abi.UintTy:
+		return "0"
+	case abi.StringTy:
+		return "\"\""
+	case abi.BoolTy:
+		return "false"
+	default:
+		return "nil"
+	}
+}
```

Comment:
Can we add a unit test to ensure that this covers every possible `abi.Type` ?

Reply:
My thinking here is that most types that are nil are `nil` and ints, strings, and bools are the only types that default to different things when `nil`. I can add a test

Reply:
The Abi types are as follows

Reply:
// Type enumerator
const (
	IntTy byte = iota
	UintTy
	BoolTy
	StringTy
	SliceTy
	ArrayTy
	TupleTy
	AddressTy
	FixedBytesTy
	BytesTy
	HashTy
	FixedPointTy
	FunctionTy
)

Reply:
All cases are covered